Provide a bill of materials (BOM)

I figured that the number of Mockito artifacts grows, so it might be useful to provide a BOM (bill of material) that lists all versions of all artifacts in a central place.

Such a thing could then be consumed in a [Gradle multi project build](https://docs.gradle.org/current/userguide/multi_project_builds.html) through

```kotlin
// my.platform.project:test
plugins {
    id("java-platform")
}

dependencies {
    api(platform("org.mockito:mockito-bom:$someVersion"))
}
```

and this could be referenced in individual module files like so:

```kotlin
dependencies {
    testImplementation(platform("my.platform.project:test"))

    testImplementation("org.mockito:mockito-core")
    testImplementation("org.mockito:mockito-junit-jupiter")
}
```
Creating and publishing such a BOM should be fairly easy. Thanks!

Done, this is what it looks like:

```

> Task :compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :javadoc
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/invocation/MatchableInvocation.java:17: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/invocation/MatchableInvocation.java:18: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/exceptions/misusing/PotentialStubbingProblem.java:29: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/exceptions/misusing/UnnecessaryStubbingException.java:33: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/exceptions/misusing/UnnecessaryStubbingException.java:34: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/exceptions/misusing/UnnecessaryStubbingException.java:42: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/ArgumentCaptor.java:18: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/ArgumentMatcher.java:83: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:54: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:54: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:55: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:55: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:73: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:252: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:252: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:264: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:528: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:683: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:710: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:735: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:747: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:761: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:767: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:932: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1079: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1079: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1203: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1203: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1208: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1213: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1217: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1217: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1231: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1236: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1261: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1261: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1261: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1265: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1277: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1277: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1289: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1289: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1289: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1293: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1293: warning - invalid usage of tag >
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1438: warning - Tag @link: reference not found: Stubbing
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1760: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:1970: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:2443: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:2457: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:2655: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:3259: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/Mockito.java:3264: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/MockitoFramework.java:36: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/MockitoFramework.java:63: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/MockSettings.java:26: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/MockSettings.java:31: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/MockSettings.java:54: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/MockSettings.java:55: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/MockSettings.java:104: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/stubbing/OngoingStubbing.java:167: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/stubbing/OngoingStubbing.java:229: warning - invalid usage of tag &
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/stubbing/VoidAnswer4.java:26: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/stubbing/VoidAnswer5.java:26: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/stubbing/VoidAnswer6.java:26: warning - invalid usage of tag <
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/exceptions/verification/opentest4j/package-info.java:9: warning - Tag @see: reference not found: org.mocktio.exceptions.verification.junit
/Users/thomaskeller/Entwicklung/src/mockito/src/main/java/org/mockito/exceptions/verification/opentest4j/package-info.java:10: warning - Tag @see: reference not found: org.mocktio.exceptions.verification.junit4
67 warnings

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 19s
35 actionable tasks: 35 executed
➜  mockito git:(bill-of-materials) ✗ ll ~/.m2/repository/org/mockito/mockito-                                                                                                                                                                    [14:22:11]
➜  mockito git:(bill-of-materials) ✗ ./gradlew publishToMavenLocal                                                                                                                                                                               [14:54:37]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

FAILURE: Build failed with an exception.

* Where:
Script '/Users/thomaskeller/Entwicklung/src/mockito/gradle/java-publication.gradle' line: 9

* What went wrong:
A problem occurred evaluating script.
> Could not get unknown property 'classes' for project ':bom' of type org.gradle.api.Project.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 5s
➜  mockito git:(bill-of-materials) ✗ ./gradlew publishToMavenLocal                                                                                                                                                                               [14:54:45]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

FAILURE: Build failed with an exception.

* Where:
Script '/Users/thomaskeller/Entwicklung/src/mockito/gradle/java-publication.gradle' line: 28

* What went wrong:
A problem occurred evaluating script.
> Could not find method jar() for arguments [java_publication_4g11eyqsjh0pm436u0aqdprwo$_run_closure3@966faf8] on project ':bom' of type org.gradle.api.Project.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 909ms
➜  mockito git:(bill-of-materials) ✗ ./gradlew publishToMavenLocal                                                                                                                                                                               [14:57:12]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

FAILURE: Build failed with an exception.

* Where:
Script '/Users/thomaskeller/Entwicklung/src/mockito/gradle/java-publication.gradle' line: 42

* What went wrong:
A problem occurred evaluating script.
> Could not get unknown property 'java' for SoftwareComponentInternal set of type org.gradle.api.internal.component.DefaultSoftwareComponentContainer.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 830ms
➜  mockito git:(bill-of-materials) ✗ ./gradlew publishToMavenLocal                                                                                                                                                                               [14:57:58]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

FAILURE: Build failed with an exception.

* Where:
Script '/Users/thomaskeller/Entwicklung/src/mockito/gradle/java-publication.gradle' line: 57

* What went wrong:
A problem occurred evaluating script.
> Could not get unknown property 'jar' for task set of type org.gradle.api.internal.tasks.DefaultTaskContainer.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 813ms
➜  mockito git:(bill-of-materials) ✗ ./gradlew publishToMavenLocal                                                                                                                                                                               [14:58:20]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

FAILURE: Build failed with an exception.

* Where:
Script '/Users/thomaskeller/Entwicklung/src/mockito/gradle/java-publication.gradle' line: 101

* What went wrong:
A problem occurred evaluating script.
> Could not get unknown property 'build' for task set of type org.gradle.api.internal.tasks.DefaultTaskContainer.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 826ms
➜  mockito git:(bill-of-materials) ✗ ./gradlew publishToMavenLocal                                                                                                                                                                               [14:58:56]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/thomaskeller/Entwicklung/src/mockito/subprojects/bom/bom.gradle' line: 9

* What went wrong:
A problem occurred evaluating project ':bom'.
> Could not get unknown property 'javadoc' for task set of type org.gradle.api.internal.tasks.DefaultTaskContainer.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 827ms
➜  mockito git:(bill-of-materials) ✗ ./gradlew publishToMavenLocal                                                                                                                                                                               [14:59:35]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/thomaskeller/Entwicklung/src/mockito/subprojects/bom/bom.gradle' line: 13

* What went wrong:
A problem occurred evaluating project ':bom'.
> Could not find method api() for arguments [project ':android'] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyConstraintHandler.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 772ms
➜  mockito git:(bill-of-materials) ✗ ./gradlew publishToMavenLocal                                                                                                                                                                               [14:59:51]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1s
37 actionable tasks: 12 executed, 25 up-to-date
➜  mockito git:(bill-of-materials) ✗ less ~/.m2/repository/org/mockito/mockito-bom/3.11.1-SNAPSHOT/mockito-bom-3.11.1-SNAPSHOT.pom                                                                                                               [15:03:51]
➜  mockito git:(bill-of-materials) ✗ less ~/.m2/repository/org/mockito/mockito-junit-jupiter/3.11.1-SNAPSHOT/mockito-junit-jupiter-3.11.1-SNAPSHOT.pom                                                                                           [15:05:07]
➜  mockito git:(bill-of-materials) ✗ ./gradlew bom:publishToMavenLocal                                                                                                                                                                           [15:05:45]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

> Configure project :bom
android
bom
deprecatedPluginsTest
errorprone
extTest
groovyTest
inline
junit-jupiter
junitJupiterExtensionTest
junitJupiterInlineMockMakerExtensionTest
junitJupiterParallelTest
kotlinReleaseCoroutinesTest
kotlinTest
memory-test
module-test
osgi-test

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1s
2 actionable tasks: 2 executed
➜  mockito git:(bill-of-materials) ✗ ./gradlew bom:publishToMavenLocal                                                                                                                                                                           [15:06:10]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

> Configure project :bom
android
errorprone
inline
junit-jupiter

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1s
2 actionable tasks: 2 executed
➜  mockito git:(bill-of-materials) ✗ less ~/.m2/repository/org/mockito/mockito-bom/3.11.1-SNAPSHOT/mockito-bom-3.11.1-SNAPSHOT.pom                                                                                                               [15:07:03]
➜  mockito git:(bill-of-materials) ✗ ./gradlew bom:publishToMavenLocal                                                                                                                                                                           [15:07:29]

> Configure project :
Building version '3.11.1-SNAPSHOT'
  - reason: shipkit-auto-version deducted snapshot based on tag: 'v3.11.0'
Spotless: customLazy has been deprecated, use custom instead

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.4.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 1s
2 actionable tasks: 2 executed
➜  mockito git:(bill-of-materials) ✗ less ~/.m2/repository/org/mockito/mockito-bom/3.11.1-SNAPSHOT/mockito-bom-3.11.1-SNAPSHOT.pom                                                                                                               [15:10:32]
➜  mockito git:(bill-of-materials) ✗ git st                                                                                                                                                                                                      [15:10:43]
On branch bill-of-materials
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
	new file:   subprojects/bom/bom.gradle

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   gradle/java-library.gradle
	modified:   gradle/java-publication.gradle
	modified:   gradle/wrapper/gradle-wrapper.properties
	modified:   settings.gradle.kts
	modified:   subprojects/bom/bom.gradle

➜  mockito git:(bill-of-materials) ✗ git add .                                                                                                                                                                                                   [15:10:44]
➜  mockito git:(bill-of-materials) ✗ git ci -am "Introduce a BOM for Mockito's artifacts (closes #2321)"                                                                                                                                         [15:10:46]
[bill-of-materials e88cb2955] Introduce a BOM for Mockito's artifacts (closes #2321)
 5 files changed, 69 insertions(+), 33 deletions(-)
 create mode 100644 subprojects/bom/bom.gradle
➜  mockito git:(bill-of-materials) git push origin HEAD                                                                                                                                                                                          [15:11:11]
Enumerating objects: 19, done.
Counting objects: 100% (19/19), done.
Delta compression using up to 8 threads
Compressing objects: 100% (10/10), done.
Writing objects: 100% (11/11), 1.38 KiB | 1.38 MiB/s, done.
Total 11 (delta 8), reused 0 (delta 0)
remote: Resolving deltas: 100% (8/8), completed with 8 local objects.
remote: 
remote: Create a pull request for 'bill-of-materials' on GitHub by visiting:
remote:      https://github.com/realdadfish/mockito/pull/new/bill-of-materials
remote: 
To github.com:realdadfish/mockito.git
 * [new branch]          HEAD -> bill-of-materials
➜  mockito git:(bill-of-materials) git reset HEAD^                                                                                                                                                                                               [15:11:17]
Unstaged changes after reset:
M	gradle/java-library.gradle
M	gradle/java-publication.gradle
M	gradle/wrapper/gradle-wrapper.properties
M	settings.gradle.kts
➜  mockito git:(bill-of-materials) ✗ git co gradle/wrapper/gradle-wrapper.properties                                                                                                                                                             [15:12:35]
Updated 1 path from the index
➜  mockito git:(bill-of-materials) ✗ git ci -am "Introduce a BOM for Mockito's artifacts (closes #2321)"                                                                                                                                         [15:12:42]
[bill-of-materials 7b5d9e81d] Introduce a BOM for Mockito's artifacts (closes #2321)
 3 files changed, 45 insertions(+), 32 deletions(-)
➜  mockito git:(bill-of-materials) ✗ git push origin HEAD -f                                                                                                                                                                                     [15:12:46]
Enumerating objects: 11, done.
Counting objects: 100% (11/11), done.
Delta compression using up to 8 threads
Compressing objects: 100% (6/6), done.
Writing objects: 100% (6/6), 880 bytes | 880.00 KiB/s, done.
Total 6 (delta 5), reused 0 (delta 0)
remote: Resolving deltas: 100% (5/5), completed with 5 local objects.
To github.com:realdadfish/mockito.git
 + e88cb2955...7b5d9e81d HEAD -> bill-of-materials (forced update)
➜  mockito git:(bill-of-materials) ✗ less ~/.m2/repository/org/mockito/mockito-bom/3.11.1-SNAPSHOT/mockito-bom-3.11.1-SNAPSHOT.pom                                                                                                               [15:12:54]

  <developers>
    <developer>
      <id>mockitoguy</id>
      <name>Szczepan Faber</name>
      <url>https://github.com/mockitoguy</url>
      <roles>
        <role>Core developer</role>
      </roles>
    </developer>
    <developer>
      <id>bric3</id>
      <name>Brice Dutheil</name>
      <url>https://github.com/bric3</url>
      <roles>
        <role>Core developer</role>
      </roles>
    </developer>
    <developer>
      <id>raphw</id>
      <name>Rafael Winterhalter</name>
      <url>https://github.com/raphw</url>
      <roles>
        <role>Core developer</role>
      </roles>
    </developer>
    <developer>
      <id>TimvdLippe</id>
      <name>Tim van der Lippe</name>
      <url>https://github.com/TimvdLippe</url>
      <roles>
        <role>Core developer</role>
      </roles>
    </developer>
  </developers>
  <scm>
    <url>https://github.com/mockito/mockito.git</url>
  </scm>
  <issueManagement>
    <system>GitHub issues</system>
    <url>https://github.com/mockito/mockito/issues</url>
  </issueManagement>
  <ciManagement>
    <system>GH Actions</system>
    <url>https://github.com/mockito/mockito/actions</url>
  </ciManagement>
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.mockito</groupId>
        <artifactId>mockito-android</artifactId>
        <version>3.11.1-SNAPSHOT</version>
      </dependency>
      <dependency>
        <groupId>org.mockito</groupId>
        <artifactId>mockito-errorprone</artifactId>
        <version>3.11.1-SNAPSHOT</version>
      </dependency>
      <dependency>
        <groupId>org.mockito</groupId>
        <artifactId>mockito-inline</artifactId>
        <version>3.11.1-SNAPSHOT</version>
      </dependency>
      <dependency>
        <groupId>org.mockito</groupId>
        <artifactId>mockito-junit-jupiter</artifactId>
        <version>3.11.1-SNAPSHOT</version>
      </dependency>
    </dependencies>
  </dependencyManagement>
</project>
```

--- a/gradle/java-library.gradle
+++ b/gradle/java-library.gradle
@@ -1,7 +1,5 @@
 apply plugin: "java"
 
-group = 'org.mockito'
-
 if (!archivesBaseName.startsWith("mockito-")) {
     archivesBaseName = "mockito-" + project.name
 }

--- a/gradle/java-publication.gradle
+++ b/gradle/java-publication.gradle
@@ -1,29 +1,34 @@
-//Sources/javadoc artifacts required by Maven module publications
-def licenseSpec = copySpec {
-    from project.rootDir
-    include "LICENSE"
+group = 'org.mockito'
+
+plugins.withId("java") {
+    //Sources/javadoc artifacts required by Maven module publications
+    def licenseSpec = copySpec {
+        from project.rootDir
+        include "LICENSE"
+    }
+
+    task sourcesJar(type: Jar, dependsOn: classes) {
+        classifier = 'sources'
+        from sourceSets.main.allSource
+        with licenseSpec
+    }
+
+    task javadocJar(type: Jar, dependsOn: javadoc) {
+        classifier = 'javadoc'
+        from tasks.javadoc
+        with licenseSpec
+    }
+
+    artifacts {
+        archives sourcesJar
+        archives javadocJar
+    }
+
+    jar {
+        with licenseSpec
+    }
 }
 
-task sourcesJar(type: Jar, dependsOn: classes) {
-    classifier = 'sources'
-    from sourceSets.main.allSource
-    with licenseSpec
-}
-
-task javadocJar(type: Jar, dependsOn: javadoc) {
-    classifier = 'javadoc'
-    from tasks.javadoc
-    with licenseSpec
-}
-
-jar {
-    with licenseSpec
-}
-
-artifacts {
-    archives sourcesJar
-    archives javadocJar
-}
 
 tasks.withType(GenerateModuleMetadata) {
     enabled = false
@@ -34,9 +39,14 @@ apply plugin: "maven-publish"
 publishing {
     publications {
         javaLibrary(MavenPublication) {
-            from components.java
-            artifact sourcesJar
-            artifact javadocJar
+            plugins.withId("java") {
+                from components.java
+                artifact sourcesJar
+                artifact javadocJar
+            }
+            plugins.withId("java-platform") {
+                from components.javaPlatform
+            }
 
             artifactId = project.archivesBaseName
 
@@ -44,10 +54,12 @@ publishing {
                 name = artifactId
                 description = project.description
 
-                //Gradle does not write 'jar' packaging to the pom (unlike other packaging types).
-                //This is OK because 'jar' is implicit/default:
-                // https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#minimal-pom
-                packaging = project.tasks.jar.extension
+                plugins.withId("java") {
+                    //Gradle does not write 'jar' packaging to the pom (unlike other packaging types).
+                    //This is OK because 'jar' is implicit/default:
+                    // https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#minimal-pom
+                    packaging = project.tasks.jar.extension
+                }
 
                 url = "https://github.com/mockito/mockito"
                 licenses {
@@ -88,8 +100,10 @@ publishing {
     repositories { maven { url = "$buildDir/repo" } }
 }
 
+plugins.withId("java") {
 //fleshes out problems with Maven pom generation when building
-tasks.build.dependsOn("publishJavaLibraryPublicationToMavenLocal")
+    tasks.build.dependsOn("publishJavaLibraryPublicationToMavenLocal")
+}
 
 apply plugin: 'signing' //https://docs.gradle.org/current/userguide/signing_plugin.html
 signing {

--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -16,7 +16,8 @@ include("deprecatedPluginsTest",
     "memory-test",
     "errorprone",
     "junitJupiterParallelTest",
-    "osgi-test")
+    "osgi-test",
+    "bom")
 
 rootProject.name = "mockito"
 

--- a/subprojects/bom/bom.gradle
+++ b/subprojects/bom/bom.gradle
@@ -1,0 +1,23 @@
+plugins {
+    id('java-platform')
+}
+
+description = "Mockito Bill of Materials (BOM)"
+
+if (!archivesBaseName.startsWith("mockito-")) {
+    archivesBaseName = "mockito-" + project.name
+}
+
+apply from: "$rootDir/gradle/java-publication.gradle"
+
+dependencies {
+    constraints {
+        project.rootProject.subprojects.forEach { subproject ->
+            if (!subproject.name.endsWith("Test") &&
+                !subproject.name.endsWith("-test") &&
+                subproject.name != "bom") {
+                api(subproject)
+            }
+        }
+    }
+}

